holidays/utils.py

```diff
@@ -318,17 +318,18 @@ def _islamic_to_gre(g_year: int, h_month: int, h_day: int) -> Iterable[date]:
     """
 
     # To avoid hijri_converter check range OverflowError.
-    dt = (g_year, h_month, h_day)
-    dt_min, dt_max = GREGORIAN_RANGE
-    if dt < dt_min or dt > dt_max:
+    min_year, max_year = (d[0] for d in GREGORIAN_RANGE)
+    if g_year <= min_year or g_year > max_year:
         return ()
 
     h_year = convert.Gregorian(g_year, 1, 1).to_hijri().year
+    min_h_year, max_h_year = (d[0] for d in HIJRI_RANGE)
+    h_years = (
+        y for y in range(h_year, h_year + 3) if min_h_year <= y <= max_h_year
```

Comment:
This one is interesting. I've had a goal to understand the lunar calendars better for a long time. And I still didn't get to a comfortable point with that.

My understanding is that we have a previous (h_year -1) and next (h_year + 2) hijri year based holidays covered at the moment. And you are replacing it with just two next year.

My question is how come it's safe to remove it (the passing tests confirm it's safe to do so or we just don't have enough cases covered)? Perhaps, a brief explanation of calculation logic would be great to add as a comment. 

I'd also like @mborsetti to look into this change as an original author (thank you!).

Reply:
> I'd also like @mborsetti to look into this change as an original author (thank you!).

@arkid15r and @KJhellico, will do over the weekend.

Reply:
> My question is how come it's safe to remove it (the passing tests confirm it's safe to do so or we just don't have enough cases covered)? Perhaps, a brief explanation of calculation logic would be great to add as a comment.

**h_year** begin before or at same day with **g_year** → **h_year-1** end strictly before **g_year** begin → there can't be days from **h_year-1** in  **g_year** → we don't need to check **h_year-1**.

**h_year+2** we need to check for holidays which falls on Hijri year begin (eg. Hijri New Year).

So, I check `range(h_year, h_year + 3)`.